homeassistant/components/climate/knx.py

```diff
@@ -246,23 +228,24 @@ def max_temp(self):
     @property
     def current_operation(self):
         """Return current operation ie. heat, cool, idle."""
-        if self.device.mode.supports_operation_mode:
-            return OPERATION_MODES.get(self.device.mode.operation_mode.value)
+        if self.device.mode is not None \
+                and self.device.mode.supports_operation_mode:
+            return self.device.mode.operation_mode.value
         return None
 
     @property
     def operation_list(self):
         """Return the list of available operation modes."""
-        return [OPERATION_MODES.get(operation_mode.value) for
+        return [operation_mode.value for
```

Comment:
Only operation modes in the climate base component are allowed.

Reply:
@MartinHjelmare KNX has a lot more than the currently supported operation modes from HA. Please see http://www.sti.uniurb.it/romanell/Domotica_e_Edifici_Intelligenti/110504-Lez10a-KNX-Datapoint%20Types%20v1.5.00%20AS.pdf for the specification. (Section 4.7 - page 111)

Another problem is that some of them are reused in multiple DPTs with different naming conventions thus making it nearly impossible to map them to the correct destination type.